servo/connectors/opsani_dev.py

```diff
@@ -106,16 +110,30 @@ def generate_kubernetes_config(
         Returns:
             A Kubernetes connector configuration object.
         """
-        main_config = servo.connectors.kubernetes.DeploymentConfiguration(
-            name=(self.deployment or self.rollout),
-            strategy=servo.connectors.kubernetes.CanaryOptimizationStrategyConfiguration(
-                type=servo.connectors.kubernetes.OptimizationStrategy.canary,
-                alias="tuning",
-            ),
-            replicas=servo.Replicas(
+        if self.create_tuning_pod:
+            strategy = (
+                servo.connectors.kubernetes.CanaryOptimizationStrategyConfiguration(
+                    type=servo.connectors.kubernetes.OptimizationStrategy.canary,
+                )
+            )
+            replicas = servo.Replicas(
                 min=0,
                 max=1,
-            ),
+            )
+
+        else:
+            strategy = servo.connectors.kubernetes.NoOptimizationStrategyConfiguration(
+                type=servo.connectors.kubernetes.OptimizationStrategy.none,
+            )
+            replicas = servo.Replicas(
+                min=0,
+                max=1000,
+            )
+
+        main_config = servo.connectors.kubernetes.DeploymentConfiguration(
+            name=(self.deployment or self.rollout),
+            strategy=strategy,
+            replicas=replicas,
```

Comment:
If we use the pinned setting with the default optimization we should be able to implement no tuning without the need for a new Optimization class in the k8s connector

```suggestion
        strategy: Union[
            servo.connectors.kubernetes.CanaryOptimizationStrategyConfiguration,
            servo.connectors.kubernetes.DefaultOptimizationStrategyConfiguration,
        ] = servo.connectors.kubernetes.DefaultOptimizationStrategyConfiguration()
        if self.create_tuning_pod:
            strategy = (
                servo.connectors.kubernetes.CanaryOptimizationStrategyConfiguration(
                    type=servo.connectors.kubernetes.OptimizationStrategy.canary,
                    alias="tuning",
                )
            )
        else:
            # NOTE: currently assuming we NEVER want to adjust the main deployment with the opsani_dev connector
            # TODO: Do we ever need to support opsani dev bootstrapping of non-canary adjusted optimization of deployments?
            self.cpu.pinned = True
            self.memory.pinned = True

        main_config = servo.connectors.kubernetes.DeploymentConfiguration(
            name=(self.deployment or self.rollout),
            strategy=strategy,
            replicas=servo.Replicas(
                min=0,
                max=1,
                pinned=True,  # NOTE always pinned for now
            ),
```

Reply:
Ok brilliant, so as I see `pinned` prevents servo modifications for the setting, but we should still need a custom replica range for non-tuning mode as this it will error if the target pods exceed this?